libsentrykube/quickpatch.py

```diff
@@ -1,15 +1,122 @@
-from typing import Sequence, Mapping
+import os
+from typing import MutableMapping, Optional, Sequence, Union
+import yaml
+import jsonpatch
+from jinja2 import Template
+
+from libsentrykube.service import (
+    get_service_path,
+    get_service_value_overrides_file_path,
+)
+from jsonschema import validate, ValidationError
+
+
+def find_patch_files(service: str, patch: str) -> Optional[str]:
+    """
+    Finds the patch file for the given service and patch name
+    """
+    base_path = get_service_path(service)
+    print(f"Base path: {base_path}")
+    for root, dirs, files in os.walk(base_path):
+        print(f"Root: {root}, dirs: {dirs}, files: {files}")
+        if "quickpatches" in root:
+            for file in files:
+                if file.endswith(f"{patch}.yaml.j2"):
+                    return os.path.join(root, file)
+    return None
+
+
+def load_pure_yaml(file_path: str) -> dict:
+    """
+    Load only the first section from the patch file that contains pure yaml and no jinja2
+    """
+    with open(file_path, "r") as file:
+        content = file.read().split("---")[
```

Comment:
Because we're using {{ }} , the yaml parser errors if it encounters this. So, we have to parse the yaml separately before replacing the {{ }} variables

Reply:
In order to validate the schema, arguments, and mappings, we need to load the yaml file. But in order to load the yaml file, we can't have {{ }}. So we can either: 
1. Load the entire yaml but replace the {{ }} with dummy arguments & resource names, validate, then re-load the entire yaml 
2. Load just the yaml without {{ }} first to validate the arguments & resource names, then use a regex to replace the {{ }} with their actual values then re-load the entire thing